client/lxd_metrics.go

```diff
@@ -0,0 +1,18 @@
+package lxd
+
+import (
+	"github.com/lxc/lxd/shared/api"
+)
+
+// GetMetrics returns metrics of the instances.
+func (r *ProtocolLXD) GetMetrics() (*api.Metrics, error) {
+	var metrics api.Metrics
+
+	// Fetch the raw value
+	_, err := r.queryStruct("GET", "/metrics", nil, "", &metrics)
```

Comment:
I'm confused, isn't that going to return us a plain-text file instead of api.Metrics?

Reply:
No, this function is called from `Metrics()` in `driver_qemu.go`. It returns the JSON representation from the `lxd-agent`. But I can see why this might be confusing.

Reply:
What we could do is add `ConnectLXDAgent` which returns an `AgentServer` which implements `InstanceServer` and also has the `GetMetrics` function. Is that too much? :D

Reply:
Actually, we could just have `ConnectLXDHTTP` return an `AgentServer` instead of `InstanceServer` as that's what we use for connecting to the agent anyway.

Reply:
We could do that. For now, could you remove this from the `client` package and call `RawQuery` from the `driver_qemu` instead to get that data?